syntax = "proto3";
package cdm;
option java_package = "com.kitware.physiology.cdm";

import "cdm/Enums.proto";
import "cdm/Properties.proto";

/** @brief Generic inhaler for substance administration. */
message InhalerData
{
  eSwitch            State        = 1;/**<< @brief Is the inhaler adminstering it's substance or not. */
  string             Substance    = 2;/**<< @brief Substance contained. */
  ScalarMassData     MeteredDose  = 3;/**<< @brief Dosage of substance. */
  Scalar0To1Data     NozzleLoss   = 4;/**<< @brief Fraction of substance lost from the inhaler. -*/
  ScalarVolumeData   SpacerVolume = 5;/**<< @brief Volume of a spacer, if used. */
}